diff --git a/src/libfetchers/github.cc b/src/libfetchers/github.cc
index 284620985..6737f196e 100644
--- a/src/libfetchers/github.cc
+++ b/src/libfetchers/github.cc
@@ -666,8 +666,81 @@ struct SourceHutInputScheme : GitArchiveInputScheme
     }
 };
 
+struct CodebergInputScheme : GitArchiveInputScheme
+{
+    std::string_view schemeName() const override { return "codeberg"; }
+
+    std::string schemeDescription() const override
+    {
+        return "";
+    }
+
+    std::optional<std::pair<std::string, std::string>> accessHeaderFromToken(const std::string & token) const override
+    {
+        return std::pair<std::string, std::string>("Authorization", fmt("token %s", token));
+    }
+
+    std::string getHost(const Input & input) const
+    {
+        return maybeGetStrAttr(input.attrs, "host").value_or("codeberg.org");
+    }
+
+    std::string getOwner(const Input & input) const
+    {
+        return getStrAttr(input.attrs, "owner");
+    }
+
+    std::string getRepo(const Input & input) const
+    {
+        return getStrAttr(input.attrs, "repo");
+    }
+
+    RefInfo getRevFromRef(const Settings & settings, nix::Store & store, const Input & input) const override
+    {
+        auto host = getHost(input);
+        auto url = fmt("https://%s/api/v1/repos/%s/%s/commits?sha=%s", host, getOwner(input), getRepo(input), *input.getRef());
+
+        Headers headers = makeHeadersWithAuthTokens(settings, host, input);
+
+        auto json = nlohmann::json::parse(
+            store.requireStoreObjectAccessor(downloadFile(store, settings, url, "source", headers).storePath)->readFile(CanonPath::root));
+
+        return RefInfo {
+            .rev = Hash::parseAny(std::string { json[1]["sha"] }, HashAlgorithm::SHA1),
+            .treeHash = Hash::parseAny(std::string { json[1]["commit"]["tree"]["sha"] }, HashAlgorithm::SHA1)
+        };
+    }
+
+    DownloadUrl getDownloadUrl(const Settings & settings, const Input & input) const override
+    {
+        auto host = getHost(input);
+
+        Headers headers = makeHeadersWithAuthTokens(settings, host, input);
+
+        // If we have no auth headers then we default to the public archive
+        // urls so we do not run into rate limits.
+        const auto urlFmt = headers.empty() ? "https://%s/%s/%s/archive/%s.tar.gz" : "https://%s/api/v1/repos/%s/%s/archive/%s.tar.gz";
+
+        const auto url = fmt(urlFmt, host, getOwner(input), getRepo(input),
+            input.getRev()->gitRev());
+
+        return DownloadUrl{parseURL(url), headers};
+    }
+
+    void clone(const Settings & settings, Store & store, const Input & input, const std::filesystem::path & destDir)
+        const override
+    {
+        auto host = getHost(input);
+        Input::fromURL(settings, fmt("git+https://%s/%s/%s.git",
+                host, getOwner(input), getRepo(input)))
+            .applyOverrides(input.getRef(), input.getRev())
+            .clone(settings, store, destDir);
+    }
+};
+
 static auto rGitHubInputScheme = OnStartup([] { registerInputScheme(std::make_unique<GitHubInputScheme>()); });
 static auto rGitLabInputScheme = OnStartup([] { registerInputScheme(std::make_unique<GitLabInputScheme>()); });
 static auto rSourceHutInputScheme = OnStartup([] { registerInputScheme(std::make_unique<SourceHutInputScheme>()); });
+static auto rCodebergInputScheme = OnStartup([] { registerInputScheme(std::make_unique<CodebergInputScheme>()); });
 
 } // namespace nix::fetchers
